gnovm/pkg/gnolang/machine.go

```diff
@@ -419,8 +419,21 @@ func (m *Machine) RunFiles(fns ...*FileNode) {
 	if pv == nil {
 		panic("RunFiles requires Machine.Package")
 	}
+	rlm := pv.GetRealm()
+	if rlm == nil && pv.IsRealm() {
+		rlm = NewRealm(pv.PkgPath) // throwaway
+	}
 	updates := m.runFileDecls(IsStdlib(pv.PkgPath), fns...)
+	if rlm != nil {
+		pb := pv.GetBlock(m.Store)
+		for _, update := range updates {
+			rlm.DidUpdate(pb, nil, update.GetFirstObject(m.Store))
+		}
```

Comment:
`update` can be a heap item, since all global variables are stored as heap item values. Therefore, a check is needed to determine whether `update` is a heap item. If it is, do not call GetFirstObject, as this would trigger a panic. thanks @leohhhn for discovering the initial bug.

Reply:
Hmm, maybe it should just work for heap items too now.
Is there a bug report or way to reproduce, or associated test that fails?
7777.